backend/config/routes.rb

```diff
@@ -5,7 +5,10 @@
   namespace :api, defaults: { format: :json } do
     resources :users, only: %i[create show destroy]
     resource :session, only: %i[create destroy]
-    resources :restaurants, only: %i[index create show]
+    resources :restaurants, only: %i[index create show] do
+      resource :favorites, only: %i[create destroy]
```

Comment:
I think this should be `resource :favorite` because it's like an action:

```
POST /api/restaurants/1/favorite
DELETE /api/restaurants/1/favorite
```

The frontend would need a change to match.